File: titan_gym_backend/src/main/java/com/example/titan_gym_backend/service/EnquiryService.java
package com.example.titan_gym_backend.service;

import com.example.titan_gym_backend.dto.request.EnquiryRequest;
import com.example.titan_gym_backend.dto.response.EnquiryResponse;

import java.util.List;

public interface EnquiryService {

    EnquiryResponse addEnquiry(EnquiryRequest request);

    List<EnquiryResponse> getAllEnquiries();

    void deleteEnquiry(Long id);

    EnquiryResponse updateStatus(Long id, String status);
}

File: gym-management-system-frontend/src/services/enquiryService.js
import API from "../api/axios";

export const addEnquiry = async (data) => {
  const response = await API.post(
    "/enquiries",
    data,
  );

  return response.data;
};

export const getAllEnquiries = async () => {
  const response = await API.get(
    "/enquiries",
  );

  return response.data;
};

export const updateStatus = async (
  id,
  status,
) => {
  const response = await API.put(
    `/enquiries/${id}/status`,
    null,
    {
      params: {
        status,
      },
    },
  );

  return response.data;
};

export const deleteEnquiry = async (id) => {
  const response = await API.delete(
    `/enquiries/${id}`,
  );

  return response.data;
};